OpenRA.Game/Settings.cs

```diff
@@ -300,6 +302,52 @@ public void Save()
 			root.WriteToFile(settingsFile);
 		}
 
+		public static string SanitizedName(string dirty)
+		{
+			// reserved characters for MiniYAML and JSON
+			var disallowedChars = new char[] { '#', '@', ':', '\n', '\t', '[', ']', '{', '}', '"', '`' };
+
+			var clean = dirty;
+			if (string.IsNullOrEmpty(dirty))
+				clean = "";
```

Comment:
Wait a moment, you are right. My ordering of the checks is wrong. Entering `#` will allow an empty name.